# Activity: Explore hypothesis testing

## Introduction

You work for an environmental think tank called Repair Our Air (ROA). ROA is formulating policy recommendations to improve the air quality in America, using the Environmental Protection Agency's Air Quality Index (AQI) to guide their decision making. An AQI value close to 0 signals "little to no" public health concern, while higher values are associated with increased risk to public health. 

They've tasked you with leveraging AQI data to help them prioritize their strategy for improving air quality in America.

ROA is considering the following decisions. For each, construct a hypothesis test and an accompanying visualization, using your results of that test to make a recommendation:

1. ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.
2. With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?
3. A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?

**Notes:**
1. For your analysis, you'll default to a 5% level of significance.
2. Throughout the lab, for two-sample t-tests, use Welch's t-test (i.e., setting the `equal_var` parameter to `False` in `scipy.stats.ttest_ind()`). This will account for the possibly unequal variances between the two groups in the comparison.

## Step 1: Imports

To proceed with your analysis, import `pandas` and `numpy`. To conduct your hypothesis testing, import `stats` from `scipy`.

#### Import Packages

In [1]:
# Import relevant packages
import numpy as np
import pandas as pd
from scipy import stats

You are also provided with a dataset with national Air Quality Index (AQI) measurements by state over time for this analysis. `Pandas` was used to import the file `c4_epa_air_quality.csv` as a dataframe named `aqi`. As shown in this cell, the dataset has been automatically loaded in for you. You do not need to download the .csv file, or provide more code, in order to access the dataset and proceed with this lab. Please continue with this activity by completing the following instructions.

**Note:** For purposes of your analysis, you can assume this data is randomly sampled from a larger population.

#### Load Dataset

In [2]:
# RUN THIS CELL TO IMPORT YOUR DATA.

aqi = pd.read_csv('c4_epa_air_quality.csv')

## Step 2: Data Exploration

### Before proceeding to your deliverables, explore your datasets.

Use the following space to surface descriptive statistics about your data. In particular, explore whether you believe the research questions you were given are readily answerable with this data.

In [3]:
# Explore your dataframe `aqi` here:
aqi.head()

Unnamed: 0  date_local    state_name   county_name      city_name  \
0           0  2018-01-01       Arizona      Maricopa        Buckeye   
1           1  2018-01-01          Ohio       Belmont      Shadyside   
2           2  2018-01-01       Wyoming         Teton  Not in a city   
3           3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4           4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158    5  
2  Parts per million         0.111111    2  
3  Parts per million         0.300000    3  
4  Parts per million         0.215789    3

In [4]:
aqi.shape

(260, 10)

In [5]:
aqi.describe()

Unnamed: 0  arithmetic_mean         aqi
count  260.000000       260.000000  260.000000
mean   129.500000         0.403169    6.757692
std     75.199734         0.317902    7.061707
min      0.000000         0.000000    0.000000
25%     64.750000         0.200000    2.000000
50%    129.500000         0.276315    5.000000
75%    194.250000         0.516009    9.000000
max    259.000000         1.921053   50.000000

In [18]:
aqi.describe(include='O')

date_local  state_name  county_name      city_name local_site_name  \
count          260         260          260            260             257   
unique           1          52          149            190             253   
top     2018-01-01  California  Los Angeles  Not in a city         Kapolei   
freq           260          66           14             21               2   

         parameter_name   units_of_measure  
count               260                260  
unique                1                  1  
top     Carbon monoxide  Parts per million  
freq                260                260

In [6]:
aqi['state_name'].value_counts()

state_name
California              66
Arizona                 14
Ohio                    12
Florida                 12
Texas                   10
Pennsylvania            10
New York                10
Colorado                 9
Michigan                 9
Minnesota                7
New Jersey               6
Indiana                  5
Nevada                   4
Maryland                 4
Massachusetts            4
Oklahoma                 4
North Carolina           4
Virginia                 4
Connecticut              4
Wyoming                  3
Utah                     3
Hawaii                   3
Illinois                 3
Missouri                 3
Iowa                     3
Kentucky                 3
Vermont                  3
Rhode Island             2
Alaska                   2
Georgia                  2
Washington               2
Idaho                    2
Nebraska                 2
Montana                  2
Tennessee                2
Maine                    2
New Mexico       

In [7]:
aqi['county_name'].value_counts()

county_name
Los Angeles              14
Wayne                     8
Maricopa                  8
San Bernardino            6
Pima                      5
                         ..
Fairbanks North Star      1
Fairfield                 1
Dallas                    1
Washington                1
Dodge                     1
Name: count, Length: 149, dtype: int64

<details>
  <summary><h4><strong>HINT 1</strong></h4></summary>

  Consider referring to the material on descriptive statisics.
</details>

<details>
  <summary><h4><strong>HINT 2</strong></h4></summary>

  Consider using `pandas` or `numpy` to explore the `aqi` dataframe.
</details>

<details>
  <summary><h4><strong>HINT 3</strong></h4></summary>

Any of the following functions may be useful:
- `pandas`: `describe()`,`value_counts()`,`shape()`, `head()`
- `numpy`: `unique()`,`mean()`
    
</details>

#### **Question 1: From the preceding data exploration, what do you recognize?**

- Our data has 260 rows and 10 columns. 2 of these columns cn be drop such as `Unnamed: 0` since this is same as our index and `date_local` since we only have 1 unique sampling date.

- States are not equally represented, as observed California alone has 66 entries while other states have as low as 1 sampling.

## Step 3. Statistical Tests

Before you proceed, recall the following steps for conducting hypothesis testing:

1. Formulate the null hypothesis and the alternative hypothesis.<br>
2. Set the significance level.<br>
3. Determine the appropriate test procedure.<br>
4. Compute the p-value.<br>
5. Draw your conclusion.

### Hypothesis 1: ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [8]:
aqi[aqi['state_name']== 'California']

Unnamed: 0  date_local  state_name     county_name         city_name  \
16           16  2018-01-01  California  San Bernardino           Ontario   
18           18  2018-01-01  California      Sacramento      Arden-Arcade   
26           26  2018-01-01  California          Orange          La Habra   
27           27  2018-01-01  California         Alameda     Not in a city   
33           33  2018-01-01  California     Los Angeles         Lancaster   
..          ...         ...         ...             ...               ...   
233         233  2018-01-01  California     Los Angeles          Glendora   
236         236  2018-01-01  California          Solano           Vallejo   
239         239  2018-01-01  California         Alameda           Oakland   
249         249  2018-01-01  California          Orange           Anaheim   
250         250  2018-01-01  California     Los Angeles  West Los Angeles   

                  local_site_name   parameter_name   units_of_measure  \
16   Ontario Near Road (Etiwanda)  Carbon monoxide  Parts per million   
18      Sacramento-Del Paso Manor  Carbon monoxide  Parts per million   
26                       La Habra  Carbon monoxide  Parts per million   
27         Berkeley- Aquatic Park  Carbon monoxide  Parts per million   
33      Lancaster-Division Street  Carbon monoxide  Parts per million   
..                            ...              ...                ...   
233                      Glendora  Carbon monoxide  Parts per million   
236                       Vallejo  Carbon monoxide  Parts per million   
239                  Oakland West  Carbon monoxide  Parts per million   
249             Anaheim Near-Road  Carbon monoxide  Parts per million   
250              West Los Angeles  Carbon monoxide  Parts per million   

     arithmetic_mean  aqi  
16          0.747368   11  
18          0.752632   16  
26          0.673684   13  
27          1.088889   15  
33          0.394737    7  
..               ...  ...  
233         0.389474    6  
236         0.620000    9  
239         0.988889   15  
249         1.015789   15  
250         0.655556   11  

[66 rows x 10 columns]

In [9]:
# Create dataframes for each sample being compared in your test

Cal_LAcounty = aqi[(aqi['state_name']== 'California') & (aqi['county_name'] == 'Los Angeles')]
Cal_LAcounty.head()

Unnamed: 0  date_local  state_name  county_name      city_name  \
33           33  2018-01-01  California  Los Angeles      Lancaster   
42           42  2018-01-01  California  Los Angeles  Santa Clarita   
61           61  2018-01-01  California  Los Angeles       Pasadena   
76           76  2018-01-01  California  Los Angeles    Los Angeles   
109         109  2018-01-01  California  Los Angeles    Los Angeles   

                   local_site_name   parameter_name   units_of_measure  \
33       Lancaster-Division Street  Carbon monoxide  Parts per million   
42                   Santa Clarita  Carbon monoxide  Parts per million   
61                        Pasadena  Carbon monoxide  Parts per million   
76                    LAX Hastings  Carbon monoxide  Parts per million   
109  Los Angeles-North Main Street  Carbon monoxide  Parts per million   

     arithmetic_mean  aqi  
33          0.394737    7  
42          0.394737    7  
61          0.789474   16  
76          0.863158   17  
109         0.994737   17

In [10]:
Cal_other = aqi[(aqi['state_name']== 'California') & (aqi['county_name'] != 'Los Angeles')]
Cal_other.head()

Unnamed: 0  date_local  state_name     county_name      city_name  \
16          16  2018-01-01  California  San Bernardino        Ontario   
18          18  2018-01-01  California      Sacramento   Arden-Arcade   
26          26  2018-01-01  California          Orange       La Habra   
27          27  2018-01-01  California         Alameda  Not in a city   
34          34  2018-01-01  California          Fresno         Fresno   

                 local_site_name   parameter_name   units_of_measure  \
16  Ontario Near Road (Etiwanda)  Carbon monoxide  Parts per million   
18     Sacramento-Del Paso Manor  Carbon monoxide  Parts per million   
26                      La Habra  Carbon monoxide  Parts per million   
27        Berkeley- Aquatic Park  Carbon monoxide  Parts per million   
34              Fresno - Garland  Carbon monoxide  Parts per million   

    arithmetic_mean  aqi  
16         0.747368   11  
18         0.752632   16  
26         0.673684   13  
27         1.088889   15  
34         1.000000   15

<details>
  <summary><h4><strong>HINT 1</strong></h4></summary>

  Consider referencing the material on subsetting dataframes.  
</details>

<details>
  <summary><h4><strong>HINT 2</strong></h4></summary>

  Consider creating two dataframes, one for Los Angeles, and one for all other California observations.
</details>

<details>
  <summary><h4><strong>HINT 3</strong></h4></summary>

For your first dataframe, filter to `county_name` of `Los Angeles`. For your second dataframe, filter to `state_name` of `Calfornia` and `county_name` not equal to `Los Angeles`.
    
</details>

#### Formulate your hypothesis:

**Formulate your null and alternative hypotheses:**

*   $H_0$: There is no difference in the mean AQI between Los Angeles County and the rest of California.
*   $H_A$: There is a difference in the mean AQI between Los Angeles County and the rest of California.


#### Set the significance level:

In [11]:
# For this analysis, the significance level is 5%

sig_level = 0.05
sig_level

0.05

#### Determine the appropriate test procedure:

Here, you are comparing the sample means between two independent samples. Therefore, you will utilize a **two-sample  𝑡-test**.

#### Compute the P-value

In [12]:
# Compute your p-value here
stats.ttest_ind(Cal_LAcounty['aqi'], Cal_other['aqi'], equal_var = False)

TtestResult(statistic=np.float64(2.1107010796372014), pvalue=np.float64(0.049839056842410995), df=np.float64(17.08246830361151))

In [13]:
stats.ttest_ind(Cal_LAcounty['aqi'], Cal_other['aqi'], equal_var = False, alternative = 'two-sided')

TtestResult(statistic=np.float64(2.1107010796372014), pvalue=np.float64(0.049839056842410995), df=np.float64(17.08246830361151))

<details>
  <summary><h4><strong>HINT 1</strong></h4></summary>

  Consider referencing the material on how to perform a two-sample t-test.
</details>

<details>
  <summary><h4><strong>HINT 2</strong></h4></summary>

  In `ttest_ind()`, a is the aqi column from our "Los Angeles" dataframe, and b is the aqi column from the "Other California" dataframe.
</details>

<details>
  <summary><h4><strong>HINT 3</strong></h4></summary>

  Be sure to set `equal_var` = False.

</details>

#### **Question 2. What is your P-value for hypothesis 1, and what does this indicate for your null hypothesis?**

The p-value is less than the significance level of 0.05, this mean that we are to reject the null hypothesis. Therefore, there is a statistical difference between the AQI value of LA county than the rest of California County.

### Hypothesis 2: With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [14]:
# Create dataframes for each sample being compared in your test

newyork = aqi[aqi['state_name']== 'New York']
ohio = aqi[aqi['state_name'] == 'Ohio']

<details>
  <summary><h4><strong>HINT 1</strong></h4></summary>

  Consider referencing the materials on subsetting dataframes.  
</details>

<details>
  <summary><h4><strong>HINT 2</strong></h4></summary>

  Consider creating two dataframes, one for New York, and one for Ohio observations.
</details>

<details>
  <summary><h4><strong>HINT 3</strong></h4></summary>

For your first dataframe, filter to `state_name` of `New York`. For your second dataframe, filter to `state_name` of `Ohio`.
    
</details>

#### Formulate your hypothesis:

**Formulate your null and alternative hypotheses:**

*   $H_0$: The mean AQI of New York is greater than or equal to that of Ohio.
*   $H_A$: The mean AQI of New York is **below** that of Ohio.


#### Significance Level (remains at 5%)

#### Determine the appropriate test procedure:

Here, you are comparing the sample means between two independent samples in one direction. Therefore, you will utilize a **two-sample  𝑡-test**.

#### Compute the P-value

In [15]:
# Compute your p-value here

t_stat, p_value = stats.ttest_ind(newyork['aqi'], ohio['aqi'], equal_var = False, alternative = 'less')

#note to self that we choose alternative less based on how we constructed oout Ha: mean NY is less than mean OH

print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

T-statistic: -2.025951038880333
P-value: 0.030446502691934683


<details>
  <summary><h4><strong>HINT 1</strong></h4></summary>

  Consider referencing the material on how to perform a two-sample t-test.
</details>

<details>
  <summary><h4><strong>HINT 2</strong></h4></summary>

  In `ttest_ind()`, a is the aqi column from the "New York" dataframe, an b is the aqi column from the "Ohio" dataframe.
</details>

<details>
  <summary><h4><strong>HINT 3</strong></h4></summary>

  You can assign `tstat`, `pvalue` to the output of `ttest_ind`. Be sure to include `alternative = less` as part of your code.  

</details>

#### **Question 3. What is your P-value for hypothesis 2, and what does this indicate for your null hypothesis?**

The p-value is 0.03 which is less than the significance level of 0.05, meaning we will reject the null hypothesis. Therefore, the mean AQI of New York is below that of Ohio.

With limited resources, the ROA should focus more on Ohio rather than on New York.

###  Hypothesis 3: A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [16]:
# Create dataframes for each sample being compared in your test

michigan = aqi[aqi['state_name'] == 'Michigan']

<details>
  <summary><h4><strong>HINT 1</strong></h4></summary>

  Consider referencing the material on subsetting dataframes.  
</details>

<details>
  <summary><h4><strong>HINT 2</strong></h4></summary>

  Consider creating one dataframe which only includes Michigan.
</details>

#### Formulate your hypothesis:

**Formulate your null and alternative hypotheses here:**

*   $H_0$: The mean AQI of Michigan is less than or equal to 10.
*   $H_A$: The mean AQI of Michigan is greater than 10.


#### Significance Level (remains at 5%)

#### Determine the appropriate test procedure:

Here, you are comparing one sample mean relative to a particular value in one direction. Therefore, you will utilize a **one-sample  𝑡-test**. 

#### Compute the P-value

In [17]:
# Compute your p-value here

t_stat, p_value = stats.ttest_1samp(michigan['aqi'], popmean = 10, alternative = 'greater')

#note to self, we choose alternative greater because on how we constructed ha: meanMichigan is greater than 10
print(f'T-statistic: {t_stat}')
print(f'P-value: {p_value}')

T-statistic: -1.7395913343286131
P-value: 0.9399405193140109


<details>
  <summary><h4><strong>HINT 1</strong></h4></summary>

  Consider referencing the material on how to perform a one-sample t-test.
</details>

<details>
  <summary><h4><strong>HINT 2</strong></h4></summary>

  In `ttest_1samp)`, you are comparing the aqi column from your Michigan data relative to 10, the new policy threshold.
</details>

<details>
  <summary><h4><strong>HINT 3</strong></h4></summary>

  You can assign `tstat`, `pvalue` to the output of `ttest_1samp`. Be sure to include `alternative = greater` as part of your code.  

</details>

#### **Question 4. What is your P-value for hypothesis 3, and what does this indicate for your null hypothesis?**

P-value is 0.93 which is greater than the significance level of 0.05, we fail-to-reject the null hypothesis. Therefore, the mean aqi of Michigan is less than or equal to 10. 

This implies that the state of Michigan will not likely be affected by the implementation of the new law.

## Step 4. Results and Evaluation

Now that you've completed your statistical tests, you can consider your hypotheses and the results you gathered.

#### **Question 5. Did your results show that the AQI in Los Angeles County was statistically different from the rest of California?**

The result showed that there is a statistical significance between the mean AQI of Los Angeles County than the rest of California state.

#### **Question 6. Did New York or Ohio have a lower AQI?**

Based on the statistical analysis, New York has lower AQI than Ohio.

#### **Question 7: Will Michigan be affected by the new policy impacting states with a mean AQI of 10 or greater?**



Michigan will not likely be affected by the new policy

# Conclusion

**What are key takeaways from this lab?**

- At 5% significance level we have concluded the following:
    - Los Angeles mean AQI is statistically different from the rest of California
    - New York have lower mean AQI than Ohio.
    - Michigan's mean AQI is less than 10.

**What would you consider presenting to your manager as part of your findings?**

- For each test (two-sample t-test, one-sample t-test)we have done, it is necessary to present the null and alternative hypotheses, show the p-value and drive a conclusion from it.


**Congratulations!** You've completed this lab. However, you may not notice a green check mark next to this item on Coursera's platform. Please continue your progress regardless of the check mark. Just click on the "save" icon at the top of this notebook to ensure your work has been logged.